From 39b19926d37ee267ae9f31c15926be01815ea3a2 Mon Sep 17 00:00:00 2001
From: Achim Gratz <Achim.Gratz@Stromeko.DE>
Date: Sun, 16 Nov 2025 12:45:38 +0100
Subject: [PATCH 402/402] fix build: libcpp/lex.cc

---
 libcpp/lex.cc | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/libcpp/lex.cc b/libcpp/lex.cc
index 63f7a5fe5b8..cba0992ac58 100644
--- a/libcpp/lex.cc
+++ b/libcpp/lex.cc
@@ -1626,6 +1626,8 @@ class unpaired_bidi_rich_location : public rich_location
      }
   };
 
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
   unpaired_bidi_rich_location (cpp_reader *pfile, location_t loc)
   : rich_location (pfile->line_table, loc, &m_custom_label)
   {
@@ -1635,6 +1637,7 @@ class unpaired_bidi_rich_location : public rich_location
 		 SHOW_RANGE_WITHOUT_CARET,
 		 &m_custom_label);
   }
+#pragma GCC diagnostic push
 
  private:
    custom_range_label m_custom_label;
-- 
2.51.0

