--- origsrc/gcc/ada/adaint.c
+++ src/gcc/ada/adaint.c
@@ -179,7 +179,9 @@
 #elif defined (__MINGW32__) || defined (__CYGWIN__)
 
 #include "mingw32.h"
+#if defined (__MINGW32__)
 #include "share.h"
+#endif
 
 /* Current code page and CCS encoding to use, set in rtinit.c.  */
 UINT __gnat_current_codepage;
--- origsrc/gcc/ada/libgnarl/s-taprop__mingw.adb
+++ src/gcc/ada/libgnarl/s-taprop__mingw.adb
@@ -741,33 +741,6 @@
         (Self_ID.Common.Compiler_Data.Pri_Stack_Info.Base'Address,
          Self_ID.Common.Compiler_Data.Pri_Stack_Info.Limit'Address);
 
-      if Self_ID.Common.Task_Image_Len > 0 then
-         declare
-            function Set_Thread_Description
-              (H : Thread_Id; Descr : Address; Length : Integer)
-               return Integer;
-            pragma
-              Import
-                (C, Set_Thread_Description, "__gnat_set_thread_description");
-
-            Nul_Terminated_Image : constant String :=
-              Self_ID.Common.Task_Image
-                (Self_ID.Common.Task_Image'First
-                 ..
-                   Self_ID.Common.Task_Image'First
-                   + Self_ID.Common.Task_Image_Len
-                   - 1)
-              & ASCII.NUL;
-
-            Result : constant Integer :=
-              Set_Thread_Description
-                (Self_ID.Common.LL.Thread,
-                 Nul_Terminated_Image'Address,
-                 Self_ID.Common.Task_Image_Len);
-         begin
-            pragma Assert (Result = 1);
-         end;
-      end if;
    end Enter_Task;
 
    -------------------


