From 4fc51db458c135538b87be04ead88dcc256f839e Mon Sep 17 00:00:00 2001
From: Evgeny Karpov <evgeny@kmaps.co>
Date: Fri, 5 Dec 2025 19:22:58 +0100
Subject: [PATCH 37/38] Add MS variadic ABI support to aarch64-pc-cygwin target

---
 gcc/config.gcc | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/gcc/config.gcc b/gcc/config.gcc
index 5fd84e35f8ad..766652ef62c3 100644
--- a/gcc/config.gcc
+++ b/gcc/config.gcc
@@ -1332,11 +1332,12 @@ aarch64-*-cygwin*)
 	tm_file="${tm_file} mingw/winnt-dll.h"
 	xm_file=i386/xm-cygwin.h
 	tmake_file="${tmake_file} aarch64/t-aarch64"
+	tmake_file="${tmake_file} aarch64/t-aarch64-mingw"
 	tmake_file="${tmake_file} mingw/t-cygming t-slibgcc"
 	target_gtfiles="$target_gtfiles \$(srcdir)/config/mingw/winnt.cc"
 	target_gtfiles="$target_gtfiles \$(srcdir)/config/mingw/winnt-dll.cc"
 	extra_options="${extra_options} mingw/cygming.opt i386/cygwin.opt"
-	extra_objs="${extra_objs} winnt.o winnt-dll.o"
+	extra_objs="${extra_objs} aarch64-abi-ms.o winnt.o winnt-dll.o"
 	c_target_objs="${c_target_objs} msformat-c.o"
 	cxx_target_objs="${cxx_target_objs} winnt-cxx.o msformat-c.o"
 	d_target_objs="${d_target_objs} cygwin-d.o"
-- 
2.51.0

