--- a/gcc/gcc.c	2011-08-14 14:15:46.614869280 -0500
+++ b/gcc/gcc.c	2011-08-14 14:16:05.757302186 -0500
@@ -98,6 +98,5 @@ compilation is specified by a string cal
 
 /* By default there is no special suffix for target executables.  */
-/* FIXME: when autoconf is fixed, remove the host check - dj */
-#if defined(TARGET_EXECUTABLE_SUFFIX) && defined(HOST_EXECUTABLE_SUFFIX)
+#if defined(TARGET_EXECUTABLE_SUFFIX)
 #define HAVE_TARGET_EXECUTABLE_SUFFIX
 #endif
